[docs]Update deploy-bare-metal.md (#11452)
apache/pulsar: site2/docs/deploy-bare-metal.md

---

> **Tips**
>

> > 1. Single-cluster Pulsar installation should be sufficient for most use cases. If you are interested in experimenting withPulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you need to run a multi-cluster Pulsar instance, see the [guide](deploy-bare-metal-multi-cluster.md).

Reviewer: ```suggestion
> 1. You can use single-cluster Pulsar installation in most use cases, such as experimenting with Pulsar or using Pulsar in a startup or in a single team. If you need to run a multi-cluster Pulsar instance, see the [guide](deploy-bare-metal-multi-cluster.md).
```

> > ### Tips

Reviewer 2: > ** Tips **
Bold font distinguishes Tips from headings when using navigation bar.

>
> 1. Single-cluster Pulsar installation should be sufficient for most use cases. If you are interested in experimenting with
> Pulsar or using Pulsar in a startup or on a single team, it is simplest to opt for a single cluster. If you need to run a multi-cluster Pulsar instance,

> > see the [guide](deploy-bare-metal-multi-cluster.md).

Reviewer 2: ```suggestion
> 1. If you are interested in testing or using Pulsar, the easiest way is to run it on a single cluster. A single-cluster Pulsar instance supports most use cases. You can also refer to this [guide](deploy-bare-metal-multi-cluster.md) to run a multi-cluster Pulsar instance.
```

The two sentences should be in one line. It's unnecessary to put them in three lines. The changes are for your reference. You can modify the sentence by yourself.

>

> > 2. If you want to use all builtin [Pulsar IO](io-overview.md) connectors in your Pulsar deployment, you need to download `apache-pulsar-io-connectors`

Reviewer 2: ```suggestion
> 2. If you want to use all built-in [Pulsar IO](io-overview.md) connectors, you need to download `apache-pulsar-io-connectors`
```

I think we can remove "in your Pulsar deployment"... You can decide whether we omit that.

> > package and install `apache-pulsar-io-connectors` under `connectors` directory in the pulsar directory on every broker node or on every function-worker node if you

Reviewer 2: Please check all incorrect \n newlines. That's not necessary. Thanks.

> > package and install `apache-pulsar-io-connectors` under `connectors` directory in the pulsar directory on every broker node or on every function-worker node if you

Reviewer 2: I suggest we use `pulsar` directory or Pulsar directory. If it's not in code, we'd better write P in Pulsar in uppercase.

> package and install `apache-pulsar-offloaders` under `offloaders` directory in the pulsar directory on every broker node. For more details of how to configure
> this feature, you can refer to the [Tiered storage cookbook](cookbooks-tiered-storage.md).

> Deploying a Pulsar cluster involves doing the following:

Reviewer 2: ```suggestion
Deploying a Pulsar cluster consists of the following steps:
```
For your reference.

1. Deploy a [ZooKeeper](#deploy-a-zookeeper-cluster) cluster (optional)  
2. Initialize [cluster metadata](#initialize-cluster-metadata)  
3. Deploy a [BookKeeper](#deploy-a-bookkeeper-cluster) cluster  
4. Deploy one or more Pulsar [brokers](#deploy-pulsar-brokers)  

## Preparation

### Requirements

Currently, Pulsar is available for 64-bit **macOS**, **Linux**, and **Windows**. To use Pulsar, you need to install 64-bit JRE/JDK 8 or later versions.

> > You can reuse existing Zookeeper clusters.

Reviewer 2: ```suggestion
> You can reuse existing ZooKeeper clusters.
```

Do we want to add > ** Tip** above this line?

Reviewer: ZooKeeper

### Requirements

Currently, Pulsar is available for 64-bit **macOS**, **Linux**, and **Windows**. To use Pulsar, you need to install 64-bit JRE/JDK 8 or later versions.

> > If you already have an existing ZooKeeper cluster and want to reuse it, you do not need to prepare  additional machines 
> > running ZooKeeper.

Reviewer: ```suggestion
> If you want to reuse an existing ZooKeeper cluster, you do not need to prepare  additional machines running ZooKeeper.
```

* A single [DNS](https://en.wikipedia.org/wiki/Domain_Name_System) name covering all of the Pulsar broker hosts

> **Note**
> 
> * Broker is only supported on 64-bit JVM.

> > * If you do not have enough machines, or you want to test Pulsar in cluster mode (and expand the cluster later), You can fully deploy pulsar on a node on which zookeeper, bookie and broker run

Reviewer: ```suggestion
> * If you do not have enough machines, or you want to test Pulsar in cluster mode (and expand the cluster later), You can fully deploy Pulsar on a node on which ZooKeeper, bookie and broker run.
```

> **Note**
>
> * Broker is only supported on 64-bit JVM.
> * If you do not have enough machines, or to try out Pulsar in cluster mode (and expand the cluster later),

> > you can deploy a full Pulsar configuration on one node, where Zookeeper, the bookie and broker are run on the same machine.

Reviewer 2: ```suggestion
> * If you do not have enough machines, or you want to test Pulsar in cluster mode (and expand the cluster later), you can deploy Pulsar completely on one node, where ZooKeeper, bookie and broker are run on the same machine.
```

'on one node' and 'on the same machine'. Is there repetition? Can you modify it?

> * If you do not have a DNS server, you can use the multi-host format in the service URL instead.
Each machine in your cluster needs to have [Java 8](https://adoptopenjdk.net/?variant=openjdk8) or [Java 11](https://adoptopenjdk.net/?variant=openjdk11) installed.

> The following is a diagram showing the basic setup:

Reviewer 2: ```suggestion
The diagram below illustrates the basic setup of a Pulsar cluster.
```

![alt-text](assets/pulsar-basic-setup.png)

> In this diagram, connecting clients need to communicate with the Pulsar cluster using a single URL. In this case, `pulsar-cluster.acme.com` abstracts over all of the message-handling brokers. Pulsar message brokers run on machines alongside BookKeeper bookies; brokers and bookies, in turn, rely on ZooKeeper.

Reviewer 2: ```suggestion
As shown in this diagram, connecting clients communicate with the Pulsar cluster using a single URL. More precisely, the URL `pulsar-cluster.acme.com` abstracts over all message-handling brokers. Pulsar message brokers run on machines alongside BookKeeper bookies; brokers and bookies, in turn, rely on ZooKeeper.
```

I am not sure about the meaning of "abstracts over all of the message-handling brokers"

### Hardware considerations

> If you deploy a Pulsar cluster, keep in mind the following basic better choices when you do the capacity planning.

Reviewer 2: ```suggestion
When deploying a Pulsar cluster, keep in mind that the capacity of your deployment might benefit from the following configuration.
```

For your reference.

#### ZooKeeper

> For machines running ZooKeeper, it is recommended to use less powerful machines or VMs. Pulsar uses ZooKeeper only for periodic coordination-related and configuration-related tasks, not for basic operations. If you run Pulsar on [Amazon Web Services](https://aws.amazon.com/) (AWS), for example, a [t2.small](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/t2-instances.html) instance might likely suffice.

Reviewer: For machines running ZooKeeper, it is recommended to use less powerful machines or VMs. Pulsar uses ZooKeeper only for periodic coordination-related and configuration-related tasks, not for basic operations. If you run Pulsar on [Amazon Web Services (AWS)](https://aws.amazon.com/), for example, a [t2.small](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/t2-instances.html)...

Reviewer: what does "instance might likely suffice" mean?

Author: It means if you are deploying pulsar on AWS, then the example provided should be suffice.

## Install the Pulsar binary package

> > You need to install the Pulsar binary package on each machine in the cluster, including machines running ZooKeeper and BookKeeper.

Reviewer 2: Shall we take it out from "Tip" and present it as "requirements"?

> To get started deploying a Pulsar cluster on bare metal, you need to download a binary tarball release in one of the following ways:

Reviewer 2: ```suggestion
To deploy a Pulsar cluster on bare metal, you need to first download a binary tarball release by one of the following ways.
```

## Install the Pulsar binary package

> > You need to install the Pulsar binary package on each machine in the cluster, including machines running [ZooKeeper](#deploy-a-zookeeper-cluster) and [BookKeeper](#deploy-a-bookkeeper-cluster).

Reviewer: ```suggestion
> You need to install the Pulsar binary package on each machine in the cluster, including machines running ZooKeeper and BookKeeper.
```

Reviewer: we don't need to add link here.
Here "ZK" and "BK" are only component, we add link to how to deploy them. It's easily found in the navigation bar.

To get started deploying a Pulsar cluster on bare metal, you need to download a binary tarball release in one of the following ways:

* By clicking on the link below directly, which automatically triggers a download:
  * <a href="pulsar:binary_release_url" download>Pulsar {{pulsar:version}} binary release</a>
* From the Pulsar [downloads page](pulsar:download_page_url)

> * From the Pulsar [releases page](https://github.com/apache/pulsar/releases/latest) on GitHub

Reviewer: well done~

> > Since Pulsar release `2.1.0-incubating`, Pulsar provides a separate binary distribution, containing all the `builtin` connectors.

Reviewer 2: probably we should check the written form of `builtin` and keep it consistent

> > If you want to enable those `builtin` connectors, you can follow the instructions as below; otherwise you can

Reviewer 2: ```suggestion
> To enable the `builtin` connectors (optional), you can follow the instructions below.
```

For your reference. You might have better ideas.

> skip this section now.

> To get started using builtin connectors, you need to download the connectors tarball release on every broker node in one of the following ways:

Reviewer 2: To use ...

> If you already have an exsiting zookeeper cluster and want to use it, you can skip this section.

> [ZooKeeper](https://zookeeper.apache.org) manages a variety of essential coordination-related and configuration-related tasks for Pulsar. To deploy a Pulsar cluster, you need to deploy ZooKeeper first. A 3-node ZooKeeper cluster is the recommended configuration. Pulsar does not make heavy use of ZooKeeper, so more lightweight machines or VMs should suffice for running ZooKeeper.

Reviewer 2: ```suggestion
[ZooKeeper](https://zookeeper.apache.org) manages a variety of essential coordination- and configuration-related tasks for Pulsar. To deploy a Pulsar cluster, you need to deploy ZooKeeper first. A 3-node ZooKeeper cluster is the recommended configuration. Pulsar does not make heavy use of ZooKeeper, so the lightweight machines or VMs should suffice for running ZooKeeper.
```

> To begin, add all ZooKeeper servers to the configuration specified in [`conf/zookeeper.conf`](reference-configuration.md#zookeeper) (in the Pulsar directory that you create [above](#install-the-pulsar-binary-package)). The following is an example:

Reviewer 2: Can you modify this sentence?
